### DISCUSSION TITLE
Improve 2FA Token Missing Logging

### DISCUSSION THREAD

**Discussion by t0rb3n** [dani-garcia/vaultwarden#4613] [Ideas]
We see a lot of 
`[error][ERROR] 2FA token not provided` in our logs without being able to map this to a specific user.

My suggestion is to improve logging in `src/api/idenitity.rs` https://github.com/dani-garcia/vaultwarden/blob/f05398a6b36891e6e979a39937f8f8eaa6360d52/src/api/identity.rs#L505
by appending the same output as password login failed, e.g. `format!("IP: {}. Username: {}.", ip.ip, username)` (Copied from https://github.com/dani-garcia/vaultwarden/blob/f05398a6b36891e6e979a39937f8f8eaa6360d52/src/api/identity.rs#L158)

So that line could become something like this: 
`None => err_json!(_json_err_twofactor(&twofactor_ids, &user.uuid, conn).await?, "2FA token not provided ", format!("IP: {}. Username: {}.", ip.ip, user.username)),` (Not sure what properties in `user` exists)

**Comment by prov3it** [comment]
I agree.
